### PR TITLE
Add Alertlogic troubleshooting submodule

### PR DESCRIPTION
**Pull request by anton-b**
### Problem
Alertlogic support people need private commands set plugin in order to execute support activities

### Resolution
Make it possible to extend base API client in the submodules
Add git submodule which will be only available from within AL

### CODE REVIEW

**Comment by anton-b** [comment]
@juantascon please take a look

File: alertlogic-cli [hunk 1]
```diff
@@ -105,12 +104,12 @@ def main():
...
         sys.exit(2)
-
-    context = parse_args()
+    services = alertlogic.dynapi.Services()
+    context = parse_args(services)
     args = context['args']
     try:
         init_logging(args)
-        result = run_command(args, context['commands'])
+        result = run_command(args, context['commands'], services)
```
Comment (juantascon) [inline alertlogic-cli]: if you are passing `services` to run_command() might as well do all the logic there so you don't need to pass both arguments, given that `commands` are a subproduct of `services`

File: alertlogic/dynapi.py [hunk 2]
```diff
@@ -26,10 +26,18 @@ class Services():
...
+
+    def add_services(self, api_services, api_data_dir):
+        for service_name in api_services:
+            filename = api_data_dir + "/" + service_name + ".json"
             service = self.parse_apidoc_file(filename)
-            self.__dict__[service_name] = service
+            if self.__dict__.has_key(service_name):
+                currentapi = self.__dict__[service_name]
+                currentapi.merge_api(service)
+                self.__dict__[service_name]._endpoints
```
Comment (juantascon) [inline alertlogic/dynapi.py]: what's the point of this line? maybe you missed an assignation or a return?

File: alertlogic-cli (hunk 1 above)
Comment (anton-b) [inline alertlogic-cli]: +

File: alertlogic/dynapi.py (hunk 2 above)
Comment (anton-b) [inline alertlogic/dynapi.py]: +, removed, just an artifact left from some testing

**Comment by anton-b** [comment]
@juantascon done please take a look

**Pull request merged**